lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,10 +37,20 @@ def to_native(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            try:
+                get_discussion_course_or_404(CourseKey.from_string(course_id), request.user)
```

Comment:
@wajeeha-khalid You'll need to update the course_overview model to include the additional data that we need.  You'll need to increment its `VERSION` number so that all previously cached values are invalidated.